### DISCUSSION TITLE
Injecting translations to `react-aria`

### DISCUSSION THREAD

**Discussion by borglind** [adobe/react-spectrum#5956] [Q&A] [answered]
I am using `useCalendar` from `react-aria` which uses the built in translations:
https://github.com/adobe/react-spectrum/tree/main/packages/%40react-aria/calendar/intl

Does anyone know if it's possible to inject or overwrite translations?
eg. adding `th-TH` for `@react-aria/calendar`.

Currently I am looking into injecting it with a `LocalizedStringDictionary` before `useCalendar` is initialised, but i am not having any luck yet.

Any ideas or pointers are appreciated!

**Comment by snowystinger** [comment, marked as answer]
Thanks for the issue! We've discussed this internally a few times. Our primary concern with opening it up would be that our string identifiers would need to become API. We'd like to support it, but we haven't found a good way to solve it yet.

**Reply by borglind** [reply]
Thanks for getting back to me and clearing it up!

What are your thoughts of adding more languages to the package? eg. if someone contributed with, let's say 30 languages, would you be reluctant because it would add more overhead to the team in regards of QA, maintenance, docs, etc?

Not sure if that would be realistic for our team, but i am just exploring options as the languages are a blocker for us atm.

**Reply by snowystinger** [reply]
I think right now, we'd be a bit reluctant for the reasons you've mentioned. I think we'd like to solve the language pack option instead, we just haven't had the time and resources to take it on yet. But thank you very much for you offer to help contribute/qa/maintain more languages, it's appreciated!

**Comment by maranomynet** [comment]
I'm hoping to start using react-aria for a project, but I notice that Icelandic is not pre-provided by `react-aria`.
...are you telling me I can't provide my own icelandic translations?
Really?
